src/org/opendatakit/briefcase/model/FormStatus.java

```diff
@@ -25,6 +25,7 @@
   private String statusString = "";
   private final StringBuilder statusHistory = new StringBuilder();
   private boolean isSuccessful = true;
+  private int historyMaxSize = 8096;
```

Comment:
Yep, that 8096 is just me screwing up my powers of two. Updating the `JEditorPane`'s entire buffer *does* require allocating new objects, at the scale of the contents. The difference between `setText` and the alternative I used is it cuts it in half by avoiding the view teardown. Using either approach really only works at scale by bounding the buffer given that the allocations are proportional to the editor content.